util/fmtutil/format.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fmtutil
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"time"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/common/model"
+
+	"github.com/prometheus/prometheus/prompb"
+)
+
+const (
+	sumStr    = "_sum"
+	countStr  = "_count"
+	bucketStr = "_bucket"
+)
+
+var MetricMetadataTypeValue = map[string]int32{
+	"UNKNOWN":        0,
+	"COUNTER":        1,
+	"GAUGE":          2,
+	"HISTOGRAM":      3,
+	"GAUGEHISTOGRAM": 4,
+	"SUMMARY":        5,
+	"INFO":           6,
+	"STATESET":       7,
+}
+
+// FormatMetrics convert metric family to a writerequest.
+func FormatMetrics(mf map[string]*dto.MetricFamily, extraLabels map[string]string) (*prompb.WriteRequest, error) {
```

Comment:
Why is it "Format" ?  To me that is something like `Printf`.
Prometheus has a concept of "exposition format", which it parses, but here you are doing almost the opposite. Why not name it `CreateWriteRequest` ?

Reply:
You right, I'm bad at naming functions..